amundsen_application/static/js/components/common/EditableText/index.tsx

```diff
@@ -144,7 +144,7 @@ class EditableText extends React.Component<
           maxLength={this.props.maxLength}
           ref={this.textAreaRef}
           defaultValue={this.state.value}
-          disabled={this.state.isDisabled}
+          disabled={!this.props.editable}
```

Comment:
As discussed offline, the disabled feature isn't related to whether or not this is editable...it blocks interaction with the component for a use case where someone tries to edit an out of date description. (See logic in `componentDidUpdate` if curious.

For this fix, we would start on the `TableDetail` page. Notice that the text component is wrapped in an `EditableSection` component. `EditableSection` will control whether or not its children are editable with its `readyOnly` property. Let's look into a fix starting there.

Reply:
You'll also want to look into `EditableSection`'s properties to help with the task of allowing the edit button to take users to the source of truth where they can edit (if it exists)